react/src/components/calculators/late-refunds/LateRefundsTableRow.tsx

```diff
@@ -0,0 +1,89 @@
+import React, {useContext} from 'react'
+import {extractFromDateString, extractToDateString, taxYearFromString, taxYearString} from '../../../config'
+
+// types
+import {LateRefundsTableRowProps, TaxYear} from '../../../interfaces'
+import SelectTaxYear from '../../helpers/formhelpers/SelectTaxYear'
+import {LateRefundsContext} from './LateRefundsContext'
+import TextInput from '../../helpers/formhelpers/TextInput'
+
+function LateRefundsTableRow(props: {
+  row: LateRefundsTableRowProps,
+  index: number,
+  printView: boolean,
+  taxYears: TaxYear[]
+}) {
+  const {
+    rows,
+    setRows,
+    activeRowId,
+    setActiveRowId,
+    errors
+  } = useContext(LateRefundsContext)
+
+  const handleTaxYearChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const ty = e.currentTarget.value
+
+    const tYObject: TaxYear = {
+      id: ty,
+      from: new Date(extractFromDateString(ty)),
+      to: new Date(extractToDateString(ty))
+    }
+
+    setRows(rows.map((cur: LateRefundsTableRowProps) =>
+      cur.id === row.id ? {...cur, taxYear: tYObject} : cur
+    ))
+  }
+
+  const handleChange = (row: LateRefundsTableRowProps, e:  React.ChangeEvent<HTMLInputElement>) => {
+    setRows(rows.map((cur: LateRefundsTableRowProps) =>
+      cur.id === row.id ? {...cur, refund: e.currentTarget.value} : cur
+    ))
+  }
+
+  const { index, row, printView, taxYears } = props
+  return (
+    <tr
+      key={row.id}

```

Comment:
this `key` is not needed here, should be on the component itself instead of here